frontend/components/albums/album_show_container.js

```diff
@@ -9,14 +9,31 @@ import { openAlbumNavDropdown,
 
 import AlbumShow from "./album_show";
 
-const mapStateToProps = (state, ownProps) => {
-
+const mapStateToProps = ({
+    entities: { currentItem, songs, playlists } // from state
+}, { params, history, currentUser // from ownProps
+}) => {
     return {
-    currentAlbum: state.entities.currentItem,
-    tracks: state.entities.songs,
-    params: ownProps.params,
-    history: ownProps.history,
-    currentUser: ownProps.currentUser,
+    currentAlbum: currentItem,
+    tracks: songs,
+    params: params,
```

Comment:
That's true. I will try to make this more clear by passing it under the key urlParams.

### For context:
#### On the back-end, I'd been leaving it bare because it is [a specific method from Rails](https://www.rubyguides.com/2019/06/rails-params/):
> How do you access this data from Rails?
> With `params`.
> Inside your controller action’s you can call `params` to access form & URL query data.
> 
> What is params, exactly?
> It’s a method that returns an `ActionController::Parameters` object, in practice it behaves a lot like a hash.
> Now.
> Let’s say that you want to read one value from this `params` hash.
> Here’s how:
> `params[:id]`
> You get the value if it exists.
> Or nil if it doesn’t.

#### On the front-end, it is also [a specific key in an object provided by React HashRouter](https://reactrouter.com/en/main/start/concepts#route-matches):

> When a route matches the URL, it's represented by a [match](https://reactrouter.com/en/main/start/concepts#match) object. A match for <Route path=":teamId" element={<Team/>}/> would look something like this:
> 
> ```
> {
>   pathname: "/teams/firebirds",
>   params: {
>     teamId: "firebirds"
>   },
>   route: {
>     element: <Team />,
>     path: ":teamId"
>   }
> }
> ```
> `pathname` holds the portion of the URL that matched this route (in our case it's all of it). `params` holds the parsed values from any [dynamic segments](https://reactrouter.com/en/main/start/concepts#dynamic-segment) that matched. Note that the param's object keys map directly to the name of the segment: `:teamId` becomes `params.teamId`.

I think this is why I'd been confused when you were telling me I couldn't just call it `params` by itself. Let me know if I'm misunderstanding something else?

Reply:
Oh I see this isn't something you chose, nvm then